modules/abstract-eth/src/abstractEthLikeNewCoins.ts

```diff
@@ -1966,44 +1989,35 @@ export abstract class AbstractEthLikeNewCoins extends AbstractEthLikeCoin {
     };
 
     const txParams = {
-      to: params.recoveryDestination, // no contract address, so this field should not be used anyways
-      nonce: backupKeyNonce,
+      to: params.recoveryDestination,
+      nonce: nonce,
       value: txAmount,
       gasPrice: gasPrice,
       gasLimit: gasLimit,
-      data: Buffer.from('0x'), // no contract call
+      data: Buffer.from('0x'),
       eip1559: params.eip1559,
       replayProtectionOptions: params.replayProtectionOptions,
     };
 
-    let tx = AbstractEthLikeNewCoins.buildTransaction(txParams);
+    const tx = AbstractEthLikeNewCoins.buildTransaction(txParams);
+    return { txInfo, tx, nonce };
+  }
 
-    if (isUnsignedSweep) {
-      return this.formatForOfflineVaultTSS(
-        txInfo,
-        tx,
-        userPublicOrPrivateKeyShare,
-        backupPrivateOrPublicKeyShare,
-        gasPrice,
-        gasLimit,
-        backupKeyNonce,
-        params.eip1559,
-        params.replayProtectionOptions
-      );
-    }
+  async validateBalanceAndGetTxAmount(baseAddress: string, gasPrice: BN, gasLimit: BN) {
+    const baseAddressBalance = await this.queryAddressBalance(baseAddress);
 
-    const signableHex = tx.getMessageToSign(false).toString('hex');
-    if (!userKeyCombined || !backupKeyCombined) {
-      throw new Error('Missing key combined shares for user or backup');
+    const totalGasNeeded = gasPrice.mul(gasLimit);
```

Comment:
```suggestion
    const totalGasCostNeeded = gasPrice.mul(gasLimit);
```
NIT: Might be the correct term

Reply:
will make this change along with the DKLS signing PR